api/src/main/java/jakarta/persistence/query/ReadQuery.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+// Contributors:
+//     Gavin King       - 4.0
+
+package jakarta.persistence.query;
+
+import jakarta.persistence.CacheRetrieveMode;
+import jakarta.persistence.CacheStoreMode;
+import jakarta.persistence.LockModeType;
+import jakarta.persistence.QueryHint;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Specifies a query expressed in the Jakarta Persistence Query Language,
+ * usually a {@code SELECT} statement, which is executed by calling
+ * {@link jakarta.persistence.Query#getResultList() getResultList()} or
+ * {@link jakarta.persistence.Query#getSingleResult() getSingleResult()}.
+ *
+ * <p> This annotation may be applied to any abstract method of any class
+ * or interface belonging to the persistence unit, or to a method of a
+ * Jakarta Data repository. The return type of the method must agree with
```

Comment:
It might be necessary to clarify the return types a bit, because obviously `List` doesn't necessarily agrees with the type returned by the query. Also, what about `Stream`? Maybe we should list a couple of supported types and specify that vendors are free to support more?

Reply:
We could spell it out a bit more, but I think it's implied that "agrees" here would mean:

1. agrees with the type returned by `getResultList()` or `getSingleResult()` when executed via JPA, and
2. agrees with the Jakarta Data spec when executed via Jakarta Data.

But sure, it would be good to say that explicitly.

Reply:
Meh. I don't know any use convincing case for `getResultStream()`. I don't know why it was even added.

Reply:
Scratch what I said. Jakarta Data allows `Stream`, so we may as well put it on the list.